lib/client/identityfile/identity.go

```diff
@@ -186,6 +215,44 @@ func writeWithNewline(w io.Writer, data []byte) error {
 	return nil
 }
 
+func writeFile(path string, data []byte, forceOverwrite bool) error {
+	if err := checkOverwrite(path, forceOverwrite); err != nil {
+		return trace.Wrap(err)
+	}
+	if err := ioutil.WriteFile(path, data, writeFileMode); err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+func checkOverwrite(path string, force bool) error {
+	// Check if destination file exists.
+	_, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		// File doesn't exist, proceed.
+		return nil
+	}
+	if err != nil {
+		// Something else went wrong, fail.
+		return trace.Wrap(err)
+	}
+	if force {
+		// File exists but we're asked not to prompt, proceed.
+		return nil
+	}
+
+	// File exists, prompt user whether to overwrite.
+	fmt.Fprintf(os.Stderr, "Destination file %q exists. Overwrite it? [y/N]: ", path)
```

Comment:
nit: this is a nice candidate for a utility function. There's at least one other [place](https://github.com/gravitational/teleport/blob/8e1865464b1736f37acab89ba7d24b4d1b18e00a/lib/client/keyagent.go#L388-L396) which uses the same code.
Also, I have not dug deep but I saw that we're setting up signal handlers which might cause minor ux issues with interrupts. Not sure if this is an issue here though.

Reply:
Good idea, added `lib/utils/prompt.Confirmation` to handle this.
PTAL